lib/numo/sparse/coo.rb

```diff
@@ -51,6 +52,34 @@ def nnz
         end
         coords
       end
+
+      def to_narray()
+        matrix = data.class.zeros(shape)
+        if ndim == 2
+          count, curr_data, curr_col, curr_row = 0, 0, 0, 0
+          while count < data.size
+            matrix[row[curr_row]][columb[curr_col]] = data[curr_data] ###fix row&columb
+            count += 1
+            curr_data += 1
+            curr_row
+            curr_col += 1
+          end
+        end
+
+        elsif ndim == 3
+        count, curr_data, curr_col = 0, 0, 0
+        curr_row, curr_dim = 0, 0
+          while count < data.size
+            matrix[dim[curr_dim]][row[curr_row]][columb[curr_col]] = data[curr_data]
+            count += 1
+            curr_data += 1
+            curr_row += 1
+            curr_col += 1
+            curr_dim += 1
+          end
+        end
+        matrix
+      end
```

Comment:
@david81982 Could you please remove the change in this file because the COO matrix is not the target of this pull-request.